components/integrations/confirm_integration/confirm_integration.tsx

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Link} from 'react-router-dom';
+
+import {browserHistory} from 'utils/browser_history';
+import {Constants, ErrorPageTypes} from 'utils/constants';
+import {getSiteURL} from 'utils/url';
+
+import {Team} from '@mattermost/types/teams';
+import {Command, IncomingWebhook, OAuthApp, OutgoingWebhook} from '@mattermost/types/integrations';
+import {IDMappedObjects} from '@mattermost/types/utilities';
+
+import CopyText from 'components/copy_text';
+import BackstageHeader from 'components/backstage/components/backstage_header.jsx';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import {Bot} from '@mattermost/types/bots';
+
+type Props = {
+    team: Team;
+    location: {search: string};
+    commands: IDMappedObjects<Command>;
+    oauthApps: IDMappedObjects<OAuthApp>;
+    incomingHooks: IDMappedObjects<IncomingWebhook>;
+    outgoingHooks: IDMappedObjects<OutgoingWebhook>;
+    bots: Record<string, Bot>;
+}
+
+const ConfirmIntegration = ({team, location, commands, oauthApps, incomingHooks, outgoingHooks, bots}: Props): JSX.Element | null => {
+    const type = (new URLSearchParams(location.search)).get('type') || '';
```

Comment:
Previously state was used to store `type` and `id` but `setState` was never used so I converted it to variable